stackit/internal/services/iaas/securitygrouprule/resource.go

```diff
@@ -0,0 +1,721 @@
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"icmp_parameters": schema.SingleNestedAttribute{
+				Description: "ICMP Parameters. These parameters should only be provided if the protocol is ICMP.",
+				Optional:    true,
+				PlanModifiers: []planmodifier.Object{
+					objectplanmodifier.RequiresReplace(),
+				},
+				Attributes: map[string]schema.Attribute{
+					"code": schema.Int64Attribute{
+						Description: "ICMP code. Can be set if the protocol is ICMP.",
+						Required:    true,
+						PlanModifiers: []planmodifier.Int64{
+							int64planmodifier.RequiresReplace(),
+						},
+						Validators: []validator.Int64{
+							int64validator.AtLeast(0),
+							int64validator.AtMost(255),
+						},
+					},
+					"type": schema.Int64Attribute{
+						Description: "ICMP type. Can be set if the protocol is ICMP.",
+						Required:    true,
+						PlanModifiers: []planmodifier.Int64{
+							int64planmodifier.RequiresReplace(),
+						},
+						Validators: []validator.Int64{
+							int64validator.AtLeast(0),
+							int64validator.AtMost(255),
+						},
+					},
+				},
+			},
+			"ip_range": schema.StringAttribute{
+				Description: "The remote IP range which the rule should match.",
+				Optional:    true,
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					validate.IP(),
+				},
+			},
+			"port_range": schema.SingleNestedAttribute{
+				Description: "The range of ports. This should only be provided if the protocol is not ICMP.",
+				Optional:    true,
+				Computed:    true,
+				PlanModifiers: []planmodifier.Object{
+					objectplanmodifier.RequiresReplace(),
+				},
+				Attributes: map[string]schema.Attribute{
+					"max": schema.Int64Attribute{
+						Description: "The maximum port number. Should be greater or equal to the minimum.",
+						Required:    true,
+						PlanModifiers: []planmodifier.Int64{
+							int64planmodifier.RequiresReplace(),
+						},
+						Validators: []validator.Int64{
+							int64validator.AtLeast(0),
+							int64validator.AtMost(65535),
+						},
+					},
+					"min": schema.Int64Attribute{
+						Description: "The minimum port number. Should be less or equal to the minimum.",
+						Required:    true,
+						PlanModifiers: []planmodifier.Int64{
+							int64planmodifier.RequiresReplace(),
+						},
+						Validators: []validator.Int64{
+							int64validator.AtLeast(0),
+							int64validator.AtMost(65535),
+						},
+					},
+				},
+			},
+			"protocol": schema.SingleNestedAttribute{
+				Description: "The internet protocol which the rule should match.",
+				Optional:    true,
+				Computed:    true,
+				PlanModifiers: []planmodifier.Object{
+					objectplanmodifier.RequiresReplace(),
+				},
+				Attributes: map[string]schema.Attribute{
```

Comment:
just confirmed with IaaS, at least one of `name` and `number` must be provided so we must add `AtLeastOneOf` object validator: https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework-validators/objectvalidator#AtLeastOneOf`

Reply:
added the validator. Also added the info in the description